text/0034-telemetry-policies.md

```diff
@@ -0,0 +1,146 @@
+
+We would like to have more telemetry to better understand our users in order to improve their usage and feel with MeiliSearch.
+
+We also use Sentry to trace the errors that occur when using MeiliSearch.
+
+### II. Motivation
+
+We need to make informed decisions based on data instead of instincts and gut feel. Lack of data does not say anything about how correct or wrong are our intuitions. Lack of data means we did not make testable hypotheses. We will now support our decisions with data.
+
+> 🔒 Note that we strictly don't want to collect private information by default (ip address, email, website url, etc..).
+
+We want to simplify the disabling of telemetry and better inform users about what we want to track.
+
+### III. Additional Materials
+N/A
+
+### IV.Explanation
+
+Following this issue and with the beginning of a product Team we asked ourselves what we wanted to do about telemetry. https://github.com/meilisearch/transplant/issues/141.
+
+We will keep telemetry enabled by default while leaving the option for users to disable it. See Impact on Documentation part.
+
+> Some new features will have telemetry data points to check the assumptions and measure objectives achievements.
+
+Each new feature specification should list data points to be collected and measured. Obviously the specification must explain why these data points are necessary for relation to the feature.
+
+We will simplify the deactivation of all our telemetries in one action. To do that, we will remove the Sentry specific flag `MEILI_NO_SENTRY` to merge its use into the `MEILI_NO_ANALYTICS` flag. This means that its activation and deactivation will be shared with analytics within the `MEILI_NO_ANALYTICS` flag. By default, Sentry metrics will stay activated.
+
+#### Current stack for analytics
+
+We are currently using Amplitude as a quantitative analytic tool to measure usage as a big picture.
+
+E.g.
+- Cohort: Servers that are running for more than seven days with the old version of MS (< v0.10.1)
+- Cohort: Servers that are running for more than seven days with the new version of MS (> v0.10.1)
+- Cohort: Servers that are running for more than 30 days with the old version of MS (< v0.10.1)
+- Cohort: Servers that are running for more than 30 days with the new version of MS (> v0.10.1)
+- Graph: The total of every telemetry messages received
+- Graph: The total of every unique id seen
+- Graph: The number of servers that are running at least since 7 days
+- Graph: The number of servers that are running at least since 30 days
+- User Composition: Device composition
+- User Composition: Version composition
+- Indexes size distribution for more than seven days (<200k not taken into account)
+- Indexes size distribution for the others
+
+#### Future Metrics
+
+##### Number of indexes per DB
```

Comment:
If I understand correctly, after v0.21 it's one index / DB as a hard rule. If that's the case, it would make this metric... less useful 😁

Reply:
@curquiza Can you confirm or deny this? It's unfortunately very possible that I misunderstood 😬

Reply:
@dichotommy Thanks for the review! 

The DB here is the MeiliSearch instance, so it will be useful.

"it's one index / DB as a hard rule" -> In the new engine it's one LMDB store / index. @curquiza  correct me if I'm wrong! 

So a DB (a MeiliSearch instance) can be composed of multiple indexes. Think of LMDB as an internal dedicated database per index. But it is totally obfuscated on the user side.